mitmproxy/protocol/http.py

```diff
@@ -308,9 +312,12 @@ def get_response():
             # > server detects timeout, disconnects
             # > read (100-n)% of large request
             # > send large request upstream
-            self.disconnect()
-            self.connect()
-            get_response()
+            try:
+                self.disconnect()
+                self.connect()
+                get_response()
+            except NotImplementedError:
```

Comment:
Yes - this needs to be improved - same reasoning as above.
If h2 fails once, it does not make sense to reconnect, since the h2 protocol is already messed up. just tear down everything and make a new connection.
I'm open for suggestions.